pkg/registry/nodemetrics/etcd/etcd.go

```diff
@@ -0,0 +1,68 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package etcd
+
+import (
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/expapi"
+	"k8s.io/kubernetes/pkg/fields"
+	"k8s.io/kubernetes/pkg/labels"
+	"k8s.io/kubernetes/pkg/registry/generic"
+	etcdgeneric "k8s.io/kubernetes/pkg/registry/generic/etcd"
+	"k8s.io/kubernetes/pkg/registry/nodemetrics"
+	"k8s.io/kubernetes/pkg/runtime"
+	"k8s.io/kubernetes/pkg/storage"
+)
+
+// REST implements a RESTStorage for node metrics against etcd.
+type REST struct {
+	*etcdgeneric.Etcd
+}
+
+// NewREST returns a RESTStorage object that will work against node metrics.
+func NewREST(s storage.Interface) *REST {
+	var prefix = "/nodemetrics"
+	store := &etcdgeneric.Etcd{
+		NewFunc: func() runtime.Object {
+			return &expapi.DerivedNodeMetrics{}
+		},
+		NewListFunc: func() runtime.Object {
+			return &expapi.DerivedNodeMetricsList{}
+		},
+		KeyRootFunc: func(ctx api.Context) string {
+			return etcdgeneric.NamespaceKeyRootFunc(ctx, prefix)
+		},
+		KeyFunc: func(ctx api.Context, name string) (string, error) {
+			return etcdgeneric.NamespaceKeyFunc(ctx, prefix, name)
```

Comment:
It should look like Nodes - (since nodes are also not namespaced).

https://github.com/kubernetes/kubernetes/blob/master/pkg/registry/minion/etcd/etcd.go

On Sep 2, 2015, at 6:03 PM, Daniel Martí <notifications@github.com> wrote:

In pkg/registry/nodemetrics/etcd/etcd.go
<https://github.com/kubernetes/kubernetes/pull/12952#discussion_r38590628>:

> +
> +// NewREST returns a RESTStorage object that will work against node metrics.
> +func NewREST(s storage.Interface) *REST {
> +	var prefix = "/nodemetrics"
> +	store := &etcdgeneric.Etcd{
> +		NewFunc: func() runtime.Object {
> +			return &expapi.DerivedNodeMetrics{}
> +		},
> +		NewListFunc: func() runtime.Object {
> +			return &expapi.DerivedNodeMetricsList{}
> +		},
> +		KeyRootFunc: func(ctx api.Context) string {
> +			return etcdgeneric.NamespaceKeyRootFunc(ctx, prefix)
> +		},
> +		KeyFunc: func(ctx api.Context, name string) (string, error) {
> +			return etcdgeneric.NamespaceKeyFunc(ctx, prefix, name)

func NoNamespaceKeyFunc(prefix string, obj runtime.Object) in pkg/storage;
should I pass it nil as the object?

—
Reply to this email directly or view it on GitHub
<https://github.com/kubernetes/kubernetes/pull/12952/files#r38590628>.